# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/cocoa/nsDragService.mm
# Commit: 384b025f56d3
# Full Hash: 384b025f56d3f87e045ecede6b1a16b71c0b5d4f
# Author: David P <daparks@mozilla.com>
# Date: 2024-07-04 16:09:52
# Regressor Bug: 1893119
# File Overlap Count: 1
# Description:
#   Bug 1893119: Part 21 - Separate nsIDragService and nsIDragSession implementations r=gstoll,geckoview-reviewers,rkraesig,win-reviewers,m_kato
#   
#   Split the class inheritance trees for nsIDragService and nsIDragSession.
#   Remember that, before the start of this patch series, the inheritance
#   diagram was:
# ==============================================================================

diff -r beeda5ef3712 -r 384b025f56d3 widget/cocoa/nsDragService.mm
--- a/widget/cocoa/nsDragService.mm	Thu Jul 04 07:48:11 2024 +0000
+++ b/widget/cocoa/nsDragService.mm	Thu Jul 04 07:48:11 2024 +0000
@@ -44,9 +44,10 @@
 // file destination callback.
 mozilla::StaticRefPtr<nsIArray> gDraggedTransferables;
 
-nsDragService::nsDragService() {}
-
-nsDragService::~nsDragService() {}
+already_AddRefed<nsIDragSession> nsDragService::CreateDragSession() {
+  RefPtr<nsIDragSession> sess = new nsDragSession();
+  return sess.forget();
+}
 
 NSImage* nsDragSession::ConstructDragImage(nsINode* aDOMNode,
                                            const Maybe<CSSIntRegion>& aRegion,
@@ -177,7 +178,7 @@
   NS_OBJC_END_TRY_BLOCK_RETURN(nil);
 }
 
-nsresult nsDragService::InvokeDragSessionImpl(
+nsresult nsDragSession::InvokeDragSessionImpl(
     nsIWidget* aWidget, nsIArray* aTransferableArray,
     const Maybe<CSSIntRegion>& aRegion, uint32_t aActionType) {
   NS_OBJC_BEGIN_TRY_BLOCK_RETURN;
@@ -250,7 +251,6 @@
   [pbItem release];
   [dragItem setDraggingFrame:localDragRect contents:image];
 
-  nsBaseDragService::StartDragSession(aWidget);
   OpenDragPopup();
 
   mNSDraggingSession = [mNativeDragView